swift/extractor/SwiftDispatcher.h

```diff
@@ -72,19 +72,28 @@ class SwiftDispatcher {
   }
 
  private:
+  // types to be supported by assignNewLabel/fetchLabel need to be listed here
+  using Store = TrapLabelStore<swift::Decl,
+                               swift::Stmt,
+                               swift::Expr,
+                               swift::Pattern,
+                               swift::TypeRepr,
+                               swift::TypeBase>;
+
   // This method gives a TRAP label for already emitted AST node.
   // If the AST node was not emitted yet, then the emission is dispatched to a corresponding
   // visitor (see `visit(T *)` methods below).
   template <typename E>
-  TrapLabel<ToTag<E>> fetchLabel(E* e) {
+  TrapLabelOf<E> fetchLabel(E* e) {
     // this is required so we avoid any recursive loop: a `fetchLabel` during the visit of `e` might
     // end up calling `fetchLabel` on `e` itself, so we want the visit of `e` to call `fetchLabel`
-    // only after having called `assignNewLabel` on `e`
-    assert(!waitingForNewLabel && "fetchLabel called before assignNewLabel");
+    // only after having called `assignNewLabel` on `e`. `index() == 0` indicates the monostate of
+    // `Store::Handle`, i.e. its default construction
+    assert(waitingForNewLabel.index() == 0 && "fetchLabel called before assignNewLabel");
```

Comment:
Optional: I think `holds_alternative<std::monostate>` would be clearer (and more resilient) than relying on the index of `monostate`?

I generally prefer the type-based accessors to the index-based accessors because they make the behaviour obvious without having to see the specific variant type (and in some cases save a fair bit of counting!) - and a side benefit of not caring about the order is that if the order changes, the user code doesn't need updating.

Reply:
done!